pkg/trace/transform/transform.go

```diff
@@ -210,14 +218,23 @@ func MarshalEvents(events ptrace.SpanEventSlice) string {
 			str.WriteString(`"attributes":{`)
 			j := 0
 			e.Attributes().Range(func(k string, v pcommon.Value) bool {
-				if j > 0 {
-					str.WriteString(",")
+				// collect the attribute only if the key is json-parseable, else drop the attribute
+				if key, err := json.Marshal(k); err == nil {
+					if j > 0 {
+						str.WriteString(",")
+					}
+					str.WriteString(string(key))
+					str.WriteString(":")
+					if val, err := json.Marshal(v.AsRaw()); err == nil {
+						str.WriteString(string(val))
+					} else {
+						log.Warnf("Trouble parsing the following attribute value, dropping: %v", v.AsString())
+						str.WriteString(`"redacted"`)
+					}
+					j++
+				} else {
+					log.Errorf("Error parsing the following attribute key on span event %v, dropping attribute: %v", e.Name(), k)
```

Comment:
One additional small nit: The incoming OTEL event that we're translating has a `DroppedAttributesCount` field. I'm not sure how often we will hit this particular log line, but in the future maybe we increment the value so the final value of `DroppedAttributesCount` in the DD Span Event is OTEL + ones that failed during our translation